java/core/src/main/java/co/worklytics/psoxy/gateway/output/ApiDataOutputUtils.java

```diff
@@ -0,0 +1,267 @@
+package co.worklytics.psoxy.gateway.output;
+
+import co.worklytics.psoxy.ControlHeader;
+import co.worklytics.psoxy.gateway.HttpEventRequest;
+import co.worklytics.psoxy.gateway.ProcessedContent;
+import com.google.api.client.http.HttpRequest;
+import com.google.api.client.http.HttpResponse;
+import lombok.AllArgsConstructor;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.commons.lang3.tuple.Pair;
+import org.apache.http.HttpHeaders;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor(onConstructor_ = {@Inject})
+public class ApiDataOutputUtils {
```

Comment:
eh, split this stuff out from `OutputUtils`, as all quite specific to "API Data" use-case